File: app/src/main/java/com/wkz/viewpager/transformer/AccordionTransformer.java
package com.wkz.viewpager.transformer;

import android.view.View;

import com.nineoldandroids.view.ViewHelper;

/**
 * @desc 像手风琴一样折叠的转换效果
 * @date 2017-06-30
 */
public class AccordionTransformer extends AbstractBaseTransformer {

    @Override
    protected void onTransform(View page, float position) {
        ViewHelper.setPivotX(page, position < 0 ? 0F : page.getWidth());
        ViewHelper.setScaleX(page, position < 0 ? 1F + position : 1F - position);
    }
}

File: view-pager-transformer/src/main/java/com/fphoenixcorneae/view_pager_transformer/AccordionTransformer.kt
package com.fphoenixcorneae.view_pager_transformer

import android.view.View
import com.nineoldandroids.view.ViewHelper

/**
 * @desc 像手风琴一样折叠的转换效果
 * @date 2017-06-30
 */
class AccordionTransformer : AbstractBaseTransformer() {
    override fun onTransform(page: View, position: Float) {
        ViewHelper.setPivotX(page, if (position < 0) 0f else page.width.toFloat())
        ViewHelper.setScaleX(page, if (position < 0) 1f + position else 1f - position)
    }
}